hints/binary-search.md

```diff
@@ -0,0 +1,39 @@
+<br>
+<details class="hint-accordion">  
+    <summary>Recommended Time & Space Complexity</summary>
+    <p>
+    You should aim for a solution with <code>O(logn)</code> time and <code>O(1)</code> space, where <code>n</code> is the size of the input array.
+    </p>
+</details>
+
+<br>
+<details class="hint-accordion">  
+    <summary>Hint 1</summary>
+    <p>
+    Can you find an algorithm that is useful when the array is sorted? Maybe other than linear seacrh.
+    </p>
+</details>
+
+<br>
+<details class="hint-accordion">  
+    <summary>Hint 2</summary>
+    <p>
+    The problem name is the name of the algorithm that we gonna use. We need to find a target value and if it does not exist in the array return <code>-1</code>. We have <code>l</code> and <code>r</code> as the boundaries of the segment of the array in which we are searching. Try building conditions to eliminate half of the search segment at each step. Maybe sorted nature of the array can be helpful.
+    </p>
+</details>
+
+<br>
+<details class="hint-accordion">  
+    <summary>Hint 3</summary>
+    <p>
+    We compare the target value with the <code>mid</code> of the segment. For example, in <code>[1, 2, 3, 4, 5]</code> and <code>target = 4</code>, <code>mid</code> is equal to <code>3</code> and we only consider the segment <code>[4, 5]</code> for the next step. Why?
```

Comment:
minor: I would split it up like this:

For example, consider the array [1, 2, 3, 4, 5] and target = 4. The mid value is 3, thus, on the next iteration we search to the right of mid. The remaining segment is [4,5].

Reply:
Yeah, this is good statement.